docs/concepts/node-agent.md

```diff
@@ -0,0 +1,63 @@
+# Gardener Node Agent
+
+The goal of the `gardener-node-agent` is to bootstrap a machine into a worker node and maintain node-specific components, which run on the node and are unmanaged by Kubernetes (e.g. the `kubelet` service, systemd units, ...).
+
+It effectively is a Kubernetes controller deployed onto the worker node.
+
+## Architecture and Basic Design
+
+![Design](./images/gardener-nodeagent-architecture.svg)
+
+This figure visualizes the overall architecture of the `gardener-node-agent`. On the left side it starts with an [`OperatingSystemConfig`](../extensions/operatingsystemconfig.md) resource (`OSC`) with a corresponding worker pool specific `cloud-config-<worker-pool>` secret being passed by reference through the userdata to a machine by the `machine-controller-manager` (MCM)
+
+On the right side the `cloud-config` secret will be extracted and used by the `gardener-node-agent` after being installed. Details on this can be found in the next section.
```

Comment:
```suggestion
On the right side, the `cloud-config` secret will be extracted and used by the `gardener-node-agent` after being installed. Details on this can be found in the next section.
```